### PR TITLE
Dat 2839

### PR DESCRIPTION
**Pull request by jacekjursza**

### CODE REVIEW

File: extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php [hunk 1]
```diff
@@ -100,6 +100,16 @@ private function handleError( $message ) {
 		return [ $renderedValue, 'markerType' => 'nowiki' ];
 	}
 
+	private function handleXmlParseError( $errors, $xmlMarkup ) {
+		$errorRenderer = new PortableInfoboxErrorRenderService( $errors );
+		if ( $this->wg->Title->getNamespace() == NS_TEMPLATE ) {
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php]: Are we sure wgTitle will always be a Title object here?

File: extensions/wikia/PortableInfobox/services/PortableInfoboxErrorRenderService.class.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		return wfMessage( 'xml-parse-error-info' )->escaped();
+	}
+
+	public function getErrorMessage( libXMLError $error ) {
+		if ( isset( $this->supportedErrors[ $error->code ] ) ) {
+			$key = $this->supportedErrors[ $error->code ];
+		} else {
+			$key = self::XML_ERR_GENERAL;
+		}
+		return wfMessage( strtolower( $key ) )->escaped();
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxErrorRenderService.class.php]: What do you think about using the naming convention of 
'xml-parse-error-' . $key so it's more visible what is the origin of the message?

File: extensions/wikia/PortableInfobox/PortableInfobox.i18n.php [hunk 3]
```diff
@@ -5,11 +5,31 @@
...
+	'xml_err_gt_required' => "Tag definition is not properly terminated",
+	'xml_err_tag_name_mismatch' => 'Opening and ending tag mismatch',
+	'xml_err_tag_not_finished' => 'Premature end of tag'
 ];
 
 $messages[ 'qqq' ] = [
 	'portable-infobox-desc' => 'Portable Infobox extension description',
 	'unimplemented-infobox-tag' => 'Error message for using unimplemented infobox tag; $1 is the tag name without pointy braces',
-	'xml-parse-error' => 'Error message for providing incorrect XML markup'
+	'xml-parse-error-info' => 'General message for parsing problem',
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/PortableInfobox.i18n.php]: Just caught this - please make sure that we have prefixes for all messages; portable-infobox-*

File: extensions/wikia/PortableInfobox/services/PortableInfoboxErrorRenderService.class.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	/* @var $templateEngine Wikia\Template\MustacheEngine */
+	private $templateEngine;
+
+	function __construct( $errorList ) {
+		$this->templateEngine = ( new Wikia\Template\MustacheEngine )
+			->setPrefix( dirname( __FILE__ ) . '/../templates' );
+		$this->errorList = $errorList;
+	}
+
+	public function renderMarkupDebugView( $sourceCode ) {
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxErrorRenderService.class.php]: This method reads quite hard. It looks like this method could call we have:
* getLinesWithErrors($sourceCode)
** that could use something along the lines of getErrorsByLine($this->errorList)
and in the end handle rendering

Comment (jacekjursza) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxErrorRenderService.class.php]: good point

**Comment by SebastianMarzjan** [comment]
:+1:

**Pull request merged**